cpp/src/ray/util/process_helper.cc

```diff
@@ -102,11 +102,7 @@ void ProcessHelper::RayStart(CoreWorkerOptions::TaskExecutionCallback callback)
   if (log_dir.empty()) {
     std::string session_dir = ConfigInternal::Instance().session_dir;
     if (session_dir.empty()) {
-      if (!global_state_accessor) {
-        global_state_accessor = ProcessHelper::GetInstance().CreateGlobalStateAccessor(
-            redis_address, ConfigInternal::Instance().redis_password);
-        RAY_CHECK(global_state_accessor->Connect()) << "Failed to connect to GCS.";
-      }
+      RAY_CHECK(global_state_accessor->Connect()) << "Failed to connect to GCS.";
```

Comment:
Should we always call `global_state_accessor->Connect()` right after the accessor is created?

Reply:
yes we should. The connect happened in the funciton `CreateGlobalStateAccessor`, so this is a duplicated one. Let's remove it.